zeppelin-web/src/app/interpreter/interpreter.controller.js

```diff
@@ -723,16 +723,21 @@ function InterpreterCtrl ($rootScope, $scope, $http, baseUrlSrv, ngToast, $timeo
   }
 
   $scope.showSparkUI = function (settingId) {
-    $http.get(baseUrlSrv.getRestApiBase() + '/interpreter/getmetainfos/' + settingId + '?propName=url')
+    $http.get(baseUrlSrv.getRestApiBase() + '/interpreter/getmetainfos/' + settingId)
       .success(function (data, status, headers, config) {
-        let url = data.body.url
-        if (!url) {
-          BootstrapDialog.alert({
-            message: 'No spark application running'
-          })
+        if (data.body === undefined || !data.body.url) {
+          if (data.body !== undefined && data.body.sparkUiEnabled === 'false') {
```

Comment:
Thanks for the contribution.

Do you think we can make this function more generic? so all other interpreter can be benefit by it.

For example, 

* `data.body.url` : URL to open on click
* `data.body.message` : Message to show on click

instead of `data.body.sparkUiEnabled`.

What do you think?